### ISSUE TITLE
@EnableJpaRepositories#excludeFilters affects @EnableJdbcRepositories

### ISSUE THREAD

**Issue by dimone-kun** [spring-projects/spring-boot#19339]
We have Spring Data JPA and Spring Data JDBC enabled in one project. JPA repositories are located in `com.example.demo.repository` package and JDBC in `com.example.demo.repository.jdbc` package. We define next config:
```java
@Configuration
@EnableJpaRepositories(
        basePackages = "com.example.demo.repository",
        excludeFilters = @ComponentScan.Filter(type = FilterType.REGEX, pattern = "com.example.demo.repository.jdbc.*")
)
@EnableJdbcRepositories(basePackages = "com.example.demo.repository.jdbc")
public class RepositoriesConfiguration {

}
```

Now, if we start our application, we'll got this exception:
```
2019-12-10 17:10:12.510  WARN 14987 --- [           main] s.c.a.AnnotationConfigApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'runner' defined in file [/home/<...>/exclude-filters-conflict-demo/build/classes/java/main/com/example/demo/Runner.class]: Unsatisfied dependency expressed through constructor parameter 1; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'com.example.demo.repository.jdbc.DemoJdbcRepository' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {}

***************************
APPLICATION FAILED TO START
***************************

Description:
Parameter 1 of constructor in com.example.demo.Runner required a bean of type 'com.example.demo.repository.jdbc.DemoJdbcRepository' that could not be found.

Action:
Consider defining a bean of type 'com.example.demo.repository.jdbc.DemoJdbcRepository' in your configuration.
```

Spring Boot 2.2.1 - 2.2.2.

2.2.0 works fine.

**Comment by wilkinsona** [comment]
I suspect that this may be a regression in Spring Data as Spring Boot doesn't really get involved in the repository scanning when you use one of Spring Data's `@Enable…Repositories` annotations. To help us to narrow down the cause, can you please create a small sample project that reproduces the behaviour you have described, working with Spring Boot 2.2.0 and failing with 2.2.1 and 2.2.2?

**Comment by dimone-kun** [comment]
Sure.
https://github.com/dimone-kun/exclude-filters-conflict-boot-demo

**Comment by wilkinsona** [comment]
Thanks for the sample, @dimone-kun.

Your problem's caused by the fix for a bug in Spring Data JDBC. In Spring Boot 2.2.0, the version of Spring Data JDBC that was used would claim a repository even if it could not be sure that it should be a JDBC repository. This was fixed in the version of Spring Data that's included in Spring Boot 2.2.1.

A side-effect of this bug fix is that you need to annotate `DemoJdbcEntity` with `@org.springframework.data.relational.core.mapping.Table` so that Spring Data JDBC knows that it should be handling `DemoJdbcRepository`. There's an info log messages that tries to point you in the right direction:

```
2019-12-17 20:25:04.324  INFO 40272 --- [           main] .RepositoryConfigurationExtensionSupport : Spring Data JDBC - Could not safely identify store assignment for repository candidate interface com.example.demo.repository.jdbc.DemoJdbcRepository. If you want this repository to be a JDBC repository, consider annotating your entities with one of these annotations: org.springframework.data.relational.core.mapping.Table.
```

There's some discussion in https://github.com/spring-projects/spring-boot/issues/19373 about what we could do to make the problem easier to identify.
